### ISSUE TITLE
L2 mode, IP assigned, but service not responding after upgrade from v.0.12.1 to v.0.13.2

### ISSUE THREAD

**Issue by sieczak-a** [metallb/metallb#2295]
### MetalLB Version

0.13.2

### Deployment method

Manifests

### Main CNI

calico v3.24.5

### Kubernetes Version

1.25.14

### Cluster Distribution

Kubeadm - vanilla kubernetes

### Describe the bug

Using metallb `v.0.12.1` in L2 mode. Loadbalancer IP responds to requests and works fine.
After upgrading to `v.0.13.2` and using CRD's loadbalancer IP is accessible but nginx service doesn't respond.
kubeproxy mode: `iptables`
kubernetes cluster: `v.1.25.14`, 3 nodes, all master nodes, without taints

I make requests from pod inside cluster to nginx external ip pod also in the same cluster. There's no networkpolicies between namespaces.
Making request using dns notation works fine: `curl -vk https://ingress-nginx-controller.ingress-nginx`

Configuration `v.0.13.2`:
```
kind: IPAddressPool
metadata:
  name: dev-pool
  namespace: metallb-system
spec:
  addresses:
    - 172.17.23.56-172.17.23.57
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: dev
  namespace: metallb-system
spec:
  ipAddressPools:
  - dev-pool
```

Configuration before CRD (`v.0.12.1`):

```
apiVersion: v1
data:
  config: |
    address-pools:
    - name: dev-pool
      protocol: layer2
      addresses:
      - 172.17.23.56-172.17.23.57
kind: ConfigMap
metadata:
  name: config
  namespace: metallb-system

```
Example service - nginx with `externalTrafficPolicy: Local`. I want to keep src IP address so I have to use `Local`
`v.0.12.1` -> nginx is reachable and responds. Service is in `externalTrafficPolicy: Local`
```
curl -vk https://172.17.23.56
* Connected to 172.17.23.56 (172.17.23.56) port 443 (#0)
<html>
<head><title>404 Not Found</title></head>
<body>
<center><h1>404 Not Found</h1></center>
<hr><center>nginx</center>
</body>
</html>

```
`v.0.13.2` -> nginx is reachable but connection is reset, only works if `externalTrafficPolicy: Cluster`

```
curl -vk https://172.17.23.56
*   Trying 172.17.23.56:443...
* TCP_NODELAY set
* Connected to 172.17.23.56 (172.17.23.56) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* OpenSSL SSL_connect: Connection reset by peer in connection to 172.17.23.56:443
* Closing connection 0
curl: (35) OpenSSL SSL_connect: Connection reset by peer in connection to 172.17.23.56:443

```

### To Reproduce

1. Upgrade from v.0.12 to v.0.13
2. Apply `IPAddressPool` and `L2Advertisement`
3. Test nginx service using `externalTrafficPolicy:` `Local` or `Cluster`

### Expected Behavior

Nginx responds even if service is in `externalTrafficPolicy: Local`

### Additional Context

Works only if src pod and dst pod are in the same physical node.

Works on v.0.11 and v.0.12, doesn't work on v.0.13 and v.0.14

Nginx service definition:
```
apiVersion: v1
kind: Service
  labels:
    app.kubernetes.io/component: controller
    app.kubernetes.io/instance: ingress-nginx
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: ingress-nginx
    app.kubernetes.io/version: 0.44.0
    helm.sh/chart: ingress-nginx-3.23.0
  name: ingress-nginx-controller
  namespace: ingress-nginx
spec:
  allocateLoadBalancerNodePorts: false
  clusterIP: 10.255.35.12
  clusterIPs:
  - 10.255.35.12
  externalTrafficPolicy: Local
  healthCheckNodePort: 32658
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  loadBalancerIP: 172.17.23.56
  ports:
  - name: http
    port: 80
    protocol: TCP
    targetPort: http
  - name: https
    port: 443
    protocol: TCP
    targetPort: https
  selector:
    app.kubernetes.io/component: controller
    app.kubernetes.io/instance: ingress-nginx
    app.kubernetes.io/name: ingress-nginx
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - ip: 172.17.23.56

```

### I've read and agree with the following

- [X] I've checked all open and closed issues and my request is not there.
- [X] I've checked all open and closed pull requests and my request is not there.

### I've read and agree with the following

- [X] I've checked all open and closed issues and my issue is not there.
- [X] This bug is reproducible when deploying MetalLB from the main branch
- [X] I have read the [troubleshooting guide](https://metallb.universe.tf/troubleshooting/) and I am still not able to make it work
- [X] I checked the logs and MetalLB is not discarding the configuration as not valid
- [X] I enabled the debug logs, collected the information required from the cluster using the [collect script](https://raw.githubusercontent.com/metallb/metallb/main/troubleshooting/collect.sh) and will attach them to the issue
- [X] I will provide the definition of my service and the related endpoint slices and attach them to this issue

**Comment by fedepaol** [comment]
please check the troubleshooting guide. 
Is arping working ? Does the packet reach the node? If that happens, metallb is innocent and everything is working as in 0.12.x

**Comment by sieczak-a** [comment]
Thank you for the fast reply. I've seen the troubleshooting guide. @fedepaol, so what could've changed if after upgrade, packets couldn't reach the destination? I've checked [changelog](https://metallb.universe.tf/release-notes/#version-0-13-2), and nothing related to L2 mode, combined with externalTrafficPolicies, is mentioned. 

If not MetalLB, then what? CNI? How to troubleshoot?

Arping works, from pod to nginx loadbalancer ip. 
```
~ # arping 172.17.23.56
ARPING 172.17.23.56 from 10.254.142.234 eth0
Unicast reply from 172.17.23.56 [MAC:ADDRESS] 0.021ms
Unicast reply from 172.17.23.56 [MAC:ADDRESS] 0.028ms
```

**Comment by fedepaol** [comment]
If a arping works and you are getting the right mac address, I think it's not a metallb issue and metallb keeps working as it was before.  You can also tcpdump on the node and verify the packet is getting there.

Did you upgrade only metallb? Can you try to roll back to 0.12x to see if that works?

**Comment by sieczak-a** [comment]
@fedepaol Yes, only MetalLB. Initially, I upgraded from 0.11x to 0.12x, and everything worked. 
However, when I upgraded to 0.13x, the problem occurred. I can confirm that after downgrading to 0.12x, the service with `externalTrafficPolicy: Local` works as expected.

I can stay on 0.12x but I wolud like stay updated in case of future Kubernetes upgrades. 
Do you have any compatibility matrix metallb - kubernetes?

**Comment by fedepaol** [comment]
this is weird, as replying to arping is literally the only thing that metallb does. The only slightly relevant thing might be this https://github.com/metallb/metallb/pull/976 , but that landed on 0.11

"MetalLB previously required the speaker to run on the same node as a pod backing a LoadBalancer, even when the ExternalTrafficPolicy was set to cluster. You may now run the MetalLB speaker on a subset of nodes, and the LoadBalancer will work for the cluster policy, regardless of where the endpoints are located. ([PR #976](https://github.com/metallb/metallb/pull/976))"

**Comment by sieczak-a** [comment]
Thank you for mentioning these issues. ETP=cluster works, but I'd like to avoid it and know src address. 
After some digging, we noticed a mess in the iptables rules created by kube-proxy; perhaps that's the case.

If I find additional info related to this case I'll reopen it.
